library/src/main/scala/scala/scalajs/js/typedarray/Float32Array.scala

```diff
@@ -18,7 +18,7 @@ class Float32Array private extends TypedArray[Float, Float32Array] {
   def this(typedArray: TypedArray[_, _]) = this()
 
   /** Creates a new Float32Array with the elements in the given array */
-  def this(array: js.Array[_]) = this()
+  def this(array: js.Iterable[_]) = this()
```

Comment:
Yes, I reread the spec and it seems what you did now is the right thing to do. The spec has a special-case for TypedArrays, and a generic case for other iterables. It does not mention Arrays specifically (those fall into the iterable case).